react/features/base/react/components/web/MultiSelectAutocomplete.js

```diff
@@ -0,0 +1,316 @@
+                    error: false
+                });
+            })
+            .catch(() => {
+                this.setState({
+                    error: true,
+                    loading: false,
+                    isOpen: false
+                });
+            });
+    }
+
+    /**
+     * Updates the selected items when a selection event occurs.
+     *
+     * @param {Object} item - The selected item.
+     * @private
+     * @returns {void}
+     */
+    _onSelectionChange(item) {
+        const existing
+            = this.state.selectedItems.find(k => k.value === item.value);
+        let selectedItems = this.state.selectedItems;
+
+        if (existing) {
+            selectedItems = selectedItems.filter(k => k !== existing);
+        } else {
+            selectedItems.push(item);
+        }
+        this.setState({
+            isOpen: false,
+            selectedItems
+        });
+
+        if (this.props.onSelectionChange) {
+            this.props.onSelectionChange(selectedItems);
+        }
+    }
+
+    /**
+     * Renders the content of this component.
+     *
+     * @returns {XML}
+     */
+    render() {
+        const shouldFitContainer = this.props.shouldFitContainer || false;
+        const shouldFocus = this.props.shouldFocus || false;
+        const isDisabled = this.props.isDisabled || false;
+        const placeholder = this.props.placeholder || '';
+        const noMatchesFound = this.props.noMatchesFound || '';
+
+        return (
+            <div>
+                <MultiSelectStateless
+                    filterValue = { this.state.filterValue }
+                    isDisabled = { isDisabled }
+                    isOpen = { this.state.isOpen }
+                    items = { this.state.items }
+                    noMatchesFound = { noMatchesFound }
+                    onFilterChange = { this._onFilterChange }
+                    onRemoved = { this._onSelectionChange }
+                    onSelected = { this._onSelectionChange }
+                    placeholder = { placeholder }
+                    selectedItems = { this.state.selectedItems }
+                    shouldFitContainer = { shouldFitContainer }
+                    shouldFocus = { shouldFocus } />
+                { this._renderLoadingIndicator() }
+                { this._renderError() }
+            </div>
+        );
+    }
+
+    /**
+     * Renders the loading indicator.
+     *
+     * @returns {*}
+     */
+    _renderLoadingIndicator() {
+        if (!(this.state.loading
+            && !this.state.items.length
+            && this.state.filterValue.length)) {
+            return null;
+        }
+
+        const content = ( // eslint-disable-line no-extra-parens
+            <div className = 'autocomplete-loading'>
+                <Spinner
+                    isCompleting = { false }
+                    size = 'medium' />
+            </div>
+        );
+
+        return (
+            <AKInlineDialog
```

Comment:
AKInlineDialog has focus behavior causing it to have a blue border on click. Do you think it's worthwhile to prevent that behavior if possible? Maybe it's not possible--I don't know.